fails:The redo statement raises a LocalJumpError if used not within block or while/for loop
